### ISSUE TITLE
Debug failure at `getCompletionEntryCodeActionsAndSourceDisplay`

### ISSUE THREAD

**Issue by phryneas** [microsoft/TypeScript#50392]
# Bug Report

### 🔎 Search Terms

Debug Error False Expression applyCompletionCommand undefined failed

### 🕗 Version & Regression Information

This has been happening for me as long as I can remember - actb told me on github that I should maybe report it :sweat_smile: 

### 💻 Code

I cannot pinpoint this behaviour to any specific code base.

### 🙁 Actual behavior

Popup with autoimport appears, I select an entry, no import is added to the file. The following error in the extension host logs:

```
[2022-08-21 17:52:06.457] [exthost] [error] [vscode.typescript-language-features] provider FAILED
[2022-08-21 17:52:06.457] [exthost] [error] Error: <semantic> TypeScript Server Error (4.7.3)
Debug Failure. False expression.
Error: Debug Failure. False expression.
    at getCompletionEntryCodeActionsAndSourceDisplay (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:132616:22)
    at Object.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:132554:30)
    at Object.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:164645:35)
    at /nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:176487:57
    at Object.mapDefined (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:605:30)
    at IpcIOSession.Session.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:176485:33)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:175124:61)
    at /nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177172:88
    at IpcIOSession.Session.executeWithRequestId (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177163:28)
    at IpcIOSession.Session.executeCommand (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177172:33)
    at IpcIOSession.Session.onMessage (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177198:35)
    at process.<anonymous> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:179833:31)
    at process.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
    at Function.create (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:718369)
    at p.dispatchResponse (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:712217)
    at p.dispatchMessage (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:711085)
    at ChildProcess.<anonymous> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:710570)
    at ChildProcess.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
[2022-08-21 17:52:07.017] [exthost] [error] Error: <semantic> TypeScript Server Error (4.7.3)
Debug Failure. False expression.
Error: Debug Failure. False expression.
    at getCompletionEntryCodeActionsAndSourceDisplay (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:132616:22)
    at Object.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:132554:30)
    at Object.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:164645:35)
    at /nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:176487:57
    at Object.mapDefined (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:605:30)
    at IpcIOSession.Session.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:176485:33)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:175124:61)
    at /nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177172:88
    at IpcIOSession.Session.executeWithRequestId (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177163:28)
    at IpcIOSession.Session.executeCommand (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177172:33)
    at IpcIOSession.Session.onMessage (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177198:35)
    at process.<anonymous> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:179833:31)
    at process.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
    at Function.create (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:718369)
    at p.dispatchResponse (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:712217)
    at p.dispatchMessage (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:711085)
    at ChildProcess.<anonymous> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:710570)
    at ChildProcess.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21) _typescript.applyCompletionCommand {"value":"vscode.typescript-language-features","_lower":"vscode.typescript-language-features"}
[2022-08-21 17:52:07.017] [exthost] [error] Error: <semantic> TypeScript Server Error (4.7.3)
Debug Failure. False expression.
Error: Debug Failure. False expression.
    at getCompletionEntryCodeActionsAndSourceDisplay (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:132616:22)
    at Object.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:132554:30)
    at Object.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:164645:35)
    at /nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:176487:57
    at Object.mapDefined (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:605:30)
    at IpcIOSession.Session.getCompletionEntryDetails (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:176485:33)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:175124:61)
    at /nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177172:88
    at IpcIOSession.Session.executeWithRequestId (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177163:28)
    at IpcIOSession.Session.executeCommand (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177172:33)
    at IpcIOSession.Session.onMessage (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:177198:35)
    at process.<anonymous> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/node_modules/typescript/lib/tsserver.js:179833:31)
    at process.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
    at Function.create (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:718369)
    at p.dispatchResponse (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:712217)
    at p.dispatchMessage (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:711085)
    at ChildProcess.<anonymous> (/nix/store/xxxl81wm2mgas7mlj671x0dwj0qjbyia-vscode-1.69.1/lib/vscode/resources/app/extensions/typescript-language-features/dist/extension.js:2:710570)
    at ChildProcess.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21) _typescript.applyCompletionCommand undefined
```

### 🙂 Expected behavior

The selected entry should be imported.

(Sorry for my brevity - I'm on vacation, got a flight to catch tomorrow and it's already late at night - those timestamps show my home timezone, not my current timezone. Still wanted to file this Issue before I forget about it.)

**Comment by DanielRosenwasser** [comment]
Do you still see the same assertion failures if you use the nightly JS/TS extension? https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next

**Comment by andrewbranch** [comment]
A repro would be great; sharing what export this was happening on _might_ be enough if it’s from a library I can install.

**Comment by phryneas** [comment]
I've just checked with the nightly extension - unfortunately it still occurs with that.

For a reproduction, I have encountered this in multiple repositories, but one where it occurs if open source, so I guess let's take that. It's not "minimal", but I honestly wouldn't know where I would start to create a minimal reproduction for this.

* clone https://github.com/reduxjs/redux-toolkit
* `yarn install`
* open `packages/toolkit/src/tests/createAsyncThunk.typetest.ts`
* try to autocomplete `configureStore`
* error ocurs
 
As a sidenote: I am using NixOS, so a slightly patched version of VSCode. I don't think how that would cause this though - and I'm pretty sure I've encountered this on my Windows machine as well. That one is just on another continent right now, so I can't verify that part :sweat_smile:
